release/release-source

```diff
@@ -78,12 +78,21 @@ autogen_or_configure()
 # Prints out an existing tarball archive for the given version
 #  $1: The tarball version
 #  $2: The directory containing tarballs
-find_tarball()
+find_and_extract_tarballs()
 {
-    local archive
-    find "$2" -maxdepth 1 -name "*-$1.tar.*" | LC_ALL=C sort | while read archive; do
-        echo "$archive"
-        break
+    local archive name
+
+    archive=
+    find "$2" -maxdepth 1 -name "*-$1.tar.*" | LC_ALL=C sort | while read name; do
+
+        tar -xf "$name" -C "$3" --strip-components=1
+	git -C "$3" reset --quiet --hard HEAD
+
+        # The first tarball is the main one
```

Comment:
This relies on numbers getting sorted before "cache". Seems fine for our purposes where we have a fixed locale, just thinking aloud.